### PR TITLE
[Compute] Support associating disk and snapshot with a disk-access resource

### PR DESCRIPTION
**Pull request by houk-ms**
**Backgroud** 

Feature Request: CLI Support for private links of managed disks #13917 

This PR is a part of supporting private links for managed disks. Private links for managed disks is a little different from the already supported private links in CLI (e.g, keyvault). It use the disk-access as an intermediate resource rather than connect with the target resource directly. Therefore, to support the private links for managed disks, we have to do the following things. This PR works to support the second step. 

> 1. Support CURD of the new resource disk-access
> 2. **Support create/update a disk/snapshot to associate it a disk-access resource**
> 3. Support private links for managed disks

**Description**  

1. This PR bumps the snapshot to API version 2020-05-01
2. This PR add two parameters for the create/update command of `az disk` and `az sanpshot`
```
--disk-access                  : Name or ID of the disk access resource for using private
                                     endpoints on disks.
--network-access-policy        : Policy for accessing the disk via network. Allowed values:
                                     AllowAll, AllowPrivate, DenyAll.
```

- Create a disk and associate it with a disk access resource.

```
az disk create -g MyResourceGroup -n MyDisk --size-gb 10 --network-access-policy AllowPrivate 
--disk-access MyDiskAccessID
```

- Update a managed disk and associate it with a disk access resource.

```
az disk update --name MyManagedDisk --resource-group MyResourceGroup --network-access-policy AllowPrivate 
--disk-access MyDiskAccessID
```

- Create a snapshot and associate it with a disk access resource.

```
az snapshot create -g MyResourceGroup -n MySnapshot --size-gb 10 --network-access-policy 
AllowPrivate --disk-access MyDiskAccessID
```

- Update a snapshot and associate it with a disk access resource.

```
az snapshot update --name MySnapshot --resource-group MyResourceGroup --network-access-policy 
AllowPrivate --disk-access MyDiskAccessID
```

### CODE REVIEW

**Comment by yonzhan** [comment]
disk-access resource

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -118,6 +118,8 @@ def load_arguments(self, _):
                        help='Encryption type. EncryptionAtRestWithPlatformKey: Disk is encrypted with XStore managed key at rest. It is the default encryption type. EncryptionAtRestWithCustomerKey: Disk is encrypted with Customer managed key at rest.')
             c.argument('disk_encryption_set', min_api='2019-07-01', help='Name or ID of disk encryption set that is used to encrypt the disk.')
             c.argument('location', help='Location. Values from: `az account list-locations`. You can configure the default location using `az configure --defaults location=<location>`. If location is not specified and no default location specified, location will be automatically set as same as the resource group.')
+            c.argument('network_access_policy', min_api='2020-05-01', help='Policy for accessing the disk via network. Allowed values: AllowAll, AllowPrivate, DenyAll.', arg_type=get_enum_type(self.get_models('NetworkAccessPolicy')))
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:121]: Will there be duplicate `Allowed values`?

Comment (houk-ms) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:121]: sorry, what does `dulpicate Allowed values` mean here?

Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:121]: I thought `arg_type=get_enum_type(self.get_models('NetworkAccessPolicy'))` would append `Allowed values` automatically.

Comment (houk-ms) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:121]: Oh, I see. It seems no duplications here, just works normally.

**Comment by yonzhan** [comment]
Compute

**Pull request merged**